### DISCUSSION TITLE
Methods to train SNN by surrogate gradient and spatial-temporal backpropagation (STBP)?

### DISCUSSION THREAD

**Discussion by CloudyDory** [lava-nc/lava-dl#167] [Q&A]
Hi, is it possible to train SNN by traditional spatial-temporal backpropagation (STBP) and surrogate gradient? For example the method proposed in: [Spatio-Temporal Backpropagation for Training High-Performance Spiking Neural Networks](https://www.frontiersin.org/articles/10.3389/fnins.2018.00331/full)? This is implemented in other SNN libraries, such as SpikingJelly and snnTorch. And it seems that the DECOLLE plugin is still different the traditional surrogate gradient approach.

**Comment by bamsumit** [comment]
[Lava-dl SLAYER](https://github.com/lava-nc/lava-dl#lavalibdlslayer) is an enhanced version of the original [SLAYER](https://papers.nips.cc/paper/2018/file/82f2b308c3b01637c607ce05f52a2fed-Paper.pdf) paper. It uses surrogate gradient and BPTT similar to STBP you mentioned.
